lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,599 @@
+		end
+		render.SetMaterial(beam_material)
+		local dist = math.Rand(0.45,0.55)
+		if(beammode)then
+			render.DrawBeam( beamstart, beamend, basesize/4, dist, dist, color)
+			render.DrawBeam( beamstart, beamend, basesize/8, dist, dist, cwh)
+		else
+			if(phase==0)then -- only the first beam is rendered in low beam mode, only a certain distance
+				local dir2 = (beamend-beamstart):GetNormal()
+				local bdist = math.min(16,beamstart:Distance(beamend))
+				local bcut = 1-(bdist / 16)
+				render.DrawBeam( beamstart, beamstart+dir2*bdist, basesize/8, dist, 1-math.Clamp(bcut,0,1)*0.5, color)
+				render.DrawBeam( beamstart, beamstart+dir2*bdist, basesize/16, dist, 1-math.Clamp(bcut,0,1)*0.5, cwh)
+			end
+		end	
+		render.SetMaterial(laser_material)
+		if((tr.HitWorld or tr.Hit) and !tr.HitSky)then
+			local reflect = tr.Entity:GetClass() == "func_reflective_glass"
+			if(reflect)then
+				local newstart = tr.HitPos
+				local dir3 = tr.Normal - 2* tr.Normal:Dot(tr.HitNormal)*tr.HitNormal	
+				LaserPointer_DrawBeam(ply,wep,newstart,dir3,color,phase+1,nil)
+			else
+				render.DrawSprite( beamend + ((EyePos() - beamend):GetNormal() * 4), basesize, basesize, color)			
+				render.DrawSprite( beamend + ((EyePos() - beamend):GetNormal() * 4), basesize/2, basesize/2, cwh)
+				render.DrawQuadEasy( beamend + ((EyePos() - beamend):GetNormal() * 4), tr.HitNormal, basesize,basesize, color,  math.Rand(0,360) )						
+				render.DrawQuadEasy( beamend + ((EyePos() - beamend):GetNormal() * 4), tr.HitNormal, basesize/2,basesize/2, cwh,  math.Rand(0,360) )						
+				--put the above 4 lines in else for if(reflect) if you want it to only place a dot on the end 
+			end
+		end
+		
+end
+
+function LaserPointer_SVBeam(ply,wep,origin,dir,phase)	-- for damage!
+	if(!SERVER or !IsValid(ply) or !IsValid(wep) or origin == nil or dir == nil)then return end
+	phase = phase or 0
+	if(phase >= 15)then return end
+	local trace = {}
+	trace.start = origin
+	trace.endpos = origin + (dir * 60000)
+	if(phase == 0)then trace.filter = ply end
+	local tr = util.TraceLine( trace )
+	if(tr.HitSky and (math.random(1,10000) == 1)) then
+	wep:MakePlane(tr.HitPos + (origin - tr.HitPos):GetNormal()*1000,ply:GetPos())
+	end 
+	if(tr.HitWorld or tr.Hit)then
+		local reflect = tr.Entity:GetClass() == "func_reflective_glass"
+		if(reflect)then
+			local newstart = tr.HitPos
+			local dir3 = tr.Normal - 2* tr.Normal:Dot(tr.HitNormal)*tr.HitNormal			
+			LaserPointer_SVBeam(ply,wep,newstart,dir3,phase+1)
+		else
+			if(IsValid(tr.Entity) and tr.Entity.Health != nil)then
+				local d = DamageInfo()
+				d:SetDamage( 1 )
+				d:SetAttacker( ply )
+				d:SetInflictor( wep )
+				d:SetDamageType( DMG_ENERGYBEAM ) 
+				tr.Entity:TakeDamageInfo( d )
+			end
+		end		
+	end
+end
+
+
+local laserhook = function(depth,skybox)
+	if(depth)then return end
+	if(skybox)then return end
+	render.SetMaterial( laser_material )
+	if(laser_material and lpointer_laser_source != nil)then
```

Comment:
lpointer_laser_source should never be nil, don't have silent-failing checks for things that shouldn't happen